alacritty/src/renderer/mod.rs

```diff
@@ -105,12 +118,17 @@ impl Renderer {
         }
 
         let (shader_version, gl_version, renderer) = unsafe {
-            let renderer = CStr::from_ptr(gl::GetString(gl::RENDERER) as *mut _).to_string_lossy();
-            let shader_version =
-                CStr::from_ptr(gl::GetString(gl::SHADING_LANGUAGE_VERSION) as *mut _)
-                    .to_string_lossy();
-            let gl_version = CStr::from_ptr(gl::GetString(gl::VERSION) as *mut _).to_string_lossy();
-            (shader_version, gl_version, renderer)
+            let shader_version = gl::GetString(gl::SHADING_LANGUAGE_VERSION);
+            let gl_version = gl::GetString(gl::VERSION);
+            let renderer = gl::GetString(gl::RENDERER);
+            if shader_version.is_null() || gl_version.is_null() || renderer.is_null() {
+                return Err("OpenGL support is missing, cannot continue".into());
+            }
+            (
+                CStr::from_ptr(shader_version as *mut _).to_string_lossy(),
+                CStr::from_ptr(gl_version as *mut _).to_string_lossy(),
+                CStr::from_ptr(renderer as *mut _).to_string_lossy(),
```

Comment:
Can't we use `glGetError`, I think it should work?

Reply:
`gl::GetError()` returns 0 at all times. `shader_version` is NULL, so it crashes in `to_string_lossy()`. `gl_version` is `1.1.0`, `renderer` is `GDI Generic`.

Reply:
I guess the driver is just straight broken, since `glGetString` returns `zero` when the `enum` is not valid, but enum variant is sort of valid, though, I guess not for the 1.1 spec, which we don't support. But still `glGetError` probably should return. though...

Reply:
Also, at which point you've called `glGetError` right after the calling `shader_version`, since otherwise it won't work, I think.

Reply:
My mistake, `gl::GetError` returns 1280 (`GL_INVALID_ENUM`) immediately after requesting `gl::SHADING_LANGUAGE_VERSION`.

Reply:
yeah, so we should propagate it.